New LAPACKFullMatrix warnings

Can someone take a look:
```
/home/fac/f/bangerth/p/deal.II/1/dealii/source/lac/lapack_full_matrix.cc:323:8: warning: unused parameter ‘A’ [-Wunused-parameter]
   void cholesky_rank1(LAPACKFullMatrix<std::complex<number>> &A,
        ^
/home/fac/f/bangerth/p/deal.II/1/dealii/source/lac/lapack_full_matrix.cc:323:8: warning: unused parameter ‘a’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/source/lac/lapack_full_matrix.cc:323:8: warning: unused parameter ‘v’ [-Wunused-parameter]
In file included from /home/fac/f/bangerth/p/deal.II/1/dealii/source/lac/lapack_full_matrix.cc:18:0:
/home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_templates.h: In instantiation of ‘void dealii::trmv(const char*, const char*, const char*, const blas_int*, const number*, const blas_int*, number*, const blas_int*) [with number = std::complex<double>; dealii::types::blas_int = int]’:
/home/fac/f/bangerth/p/deal.II/1/dealii/source/lac/lapack_full_matrix.cc:412:25:   required from ‘void dealii::LAPACKFullMatrix<number>::vmult(dealii::Vector<OtherNumber>&, const dealii::Vector<OtherNumber>&, bool) const [with number = std::complex<double>]’
source/lac/lapack_full_matrix.inst:20:16:   required from here
/home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_templates.h:426:1: warning: unused parameter ‘uplo’ [-Wunused-parameter]
 trmv (const char *uplo, const char *trans, const char *diag,
 ^
/home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_templates.h:426:1: warning: unused parameter ‘trans’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_templates.h:426:1: warning: unused parameter ‘diag’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_templates.h:426:1: warning: unused parameter ‘N’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_templates.h:426:1: warning: unused parameter ‘A’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_templates.h:426:1: warning: unused parameter ‘lda’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_templates.h:426:1: warning: unused parameter ‘x’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_templates.h:426:1: warning: unused parameter ‘incx’ [-Wunused-parameter]
In file included from /home/fac/f/bangerth/p/deal.II/1/dealii/source/lac/lapack_full_matrix.cc:18:0:
/home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_templates.h: In instantiation of ‘void dealii::trcon(const char*, const char*, const char*, const blas_int*, const number*, const blas_int*, number*, number*, dealii::types::blas_int*, dealii::types::blas_int*) [with number = std::complex<double>; dealii::types::blas_int = int]’:
/home/fac/f/bangerth/p/deal.II/1/dealii/source/lac/lapack_full_matrix.cc:953:41:   required from ‘number dealii::LAPACKFullMatrix<number>::reciprocal_condition_number() const [with number = std::complex<double>]’
source/lac/lapack_full_matrix.inst:20:16:   required from here
/home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_templates.h:556:1: warning: unused parameter ‘norm’ [-Wunused-parameter]
 trcon(const char *norm, const char *uplo, const char *diag,
 ^
/home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_templates.h:556:1: warning: unused parameter ‘uplo’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_templates.h:556:1: warning: unused parameter ‘diag’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_templates.h:556:1: warning: unused parameter ‘n’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_templates.h:556:1: warning: unused parameter ‘A’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_templates.h:556:1: warning: unused parameter ‘lda’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_templates.h:556:1: warning: unused parameter ‘rcond’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_templates.h:556:1: warning: unused parameter ‘work’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_templates.h:556:1: warning: unused parameter ‘iwork’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_templates.h:556:1: warning: unused parameter ‘info’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/source/lac/lapack_full_matrix.cc: In instantiation of ‘void dealii::{anonymous}::cholesky_rank1(dealii::LAPACKFullMatrix<std::complex<_Tp> >&, std::complex<_Tp>, const dealii::Vector<std::complex<_Tp> >&) [with number = float]’:
/home/fac/f/bangerth/p/deal.II/1/dealii/source/lac/lapack_full_matrix.cc:367:33:   required from ‘void dealii::LAPACKFullMatrix<number>::rank1_update(number, const dealii::Vector<OtherNumber>&) [with number = std::complex<float>]’
source/lac/lapack_full_matrix.inst:25:16:   required from here
/home/fac/f/bangerth/p/deal.II/1/dealii/source/lac/lapack_full_matrix.cc:323:8: warning: unused parameter ‘A’ [-Wunused-parameter]
   void cholesky_rank1(LAPACKFullMatrix<std::complex<number>> &A,
        ^
/home/fac/f/bangerth/p/deal.II/1/dealii/source/lac/lapack_full_matrix.cc:323:8: warning: unused parameter ‘a’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/source/lac/lapack_full_matrix.cc:323:8: warning: unused parameter ‘v’ [-Wunused-parameter]
In file included from /home/fac/f/bangerth/p/deal.II/1/dealii/source/lac/lapack_full_matrix.cc:18:0:
/home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_templates.h: In instantiation of ‘void dealii::trmv(const char*, const char*, const char*, const blas_int*, const number*, const blas_int*, number*, const blas_int*) [with number = std::complex<float>; dealii::types::blas_int = int]’:
/home/fac/f/bangerth/p/deal.II/1/dealii/source/lac/lapack_full_matrix.cc:412:25:   required from ‘void dealii::LAPACKFullMatrix<number>::vmult(dealii::Vector<OtherNumber>&, const dealii::Vector<OtherNumber>&, bool) const [with number = std::complex<float>]’
source/lac/lapack_full_matrix.inst:25:16:   required from here
/home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_templates.h:426:1: warning: unused parameter ‘uplo’ [-Wunused-parameter]
 trmv (const char *uplo, const char *trans, const char *diag,
 ^
/home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_templates.h:426:1: warning: unused parameter ‘trans’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_templates.h:426:1: warning: unused parameter ‘diag’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_templates.h:426:1: warning: unused parameter ‘N’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_templates.h:426:1: warning: unused parameter ‘A’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_templates.h:426:1: warning: unused parameter ‘lda’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_templates.h:426:1: warning: unused parameter ‘x’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_templates.h:426:1: warning: unused parameter ‘incx’ [-Wunused-parameter]
In file included from /home/fac/f/bangerth/p/deal.II/1/dealii/source/lac/lapack_full_matrix.cc:18:0:
/home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_templates.h: In instantiation of ‘void dealii::trcon(const char*, const char*, const char*, const blas_int*, const number*, const blas_int*, number*, number*, dealii::types::blas_int*, dealii::types::blas_int*) [with number = std::complex<float>; dealii::types::blas_int = int]’:
/home/fac/f/bangerth/p/deal.II/1/dealii/source/lac/lapack_full_matrix.cc:953:41:   required from ‘number dealii::LAPACKFullMatrix<number>::reciprocal_condition_number() const [with number = std::complex<float>]’
source/lac/lapack_full_matrix.inst:25:16:   required from here
/home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_templates.h:556:1: warning: unused parameter ‘norm’ [-Wunused-parameter]
 trcon(const char *norm, const char *uplo, const char *diag,
 ^
/home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_templates.h:556:1: warning: unused parameter ‘uplo’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_templates.h:556:1: warning: unused parameter ‘diag’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_templates.h:556:1: warning: unused parameter ‘n’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_templates.h:556:1: warning: unused parameter ‘A’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_templates.h:556:1: warning: unused parameter ‘lda’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_templates.h:556:1: warning: unused parameter ‘rcond’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_templates.h:556:1: warning: unused parameter ‘work’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_templates.h:556:1: warning: unused parameter ‘iwork’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/include/deal.II/lac/lapack_templates.h:556:1: warning: unused parameter ‘info’ [-Wunused-parameter]
[206/772] Building CXX object source/lac/CMakeFiles/obj_lac_debug.dir/sparsity_tools.cc.o
/home/fac/f/bangerth/p/deal.II/1/dealii/source/lac/sparsity_tools.cc:153:10: warning: unused parameter ‘sizeGID’ [-Wunused-parameter]
     void get_object_list(void *data, int sizeGID, int sizeLID,
          ^
/home/fac/f/bangerth/p/deal.II/1/dealii/source/lac/sparsity_tools.cc:153:10: warning: unused parameter ‘sizeLID’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/source/lac/sparsity_tools.cc:153:10: warning: unused parameter ‘wgt_dim’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/source/lac/sparsity_tools.cc:153:10: warning: unused parameter ‘obj_wgts’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/source/lac/sparsity_tools.cc:174:10: warning: unused parameter ‘sizeGID’ [-Wunused-parameter]
     void get_num_edges_list(void *data, int sizeGID, int sizeLID,
          ^
/home/fac/f/bangerth/p/deal.II/1/dealii/source/lac/sparsity_tools.cc:174:10: warning: unused parameter ‘sizeLID’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/source/lac/sparsity_tools.cc:174:10: warning: unused parameter ‘localID’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/source/lac/sparsity_tools.cc:196:10: warning: unused parameter ‘sizeGID’ [-Wunused-parameter]
     void get_edge_list(void *data, int sizeGID, int sizeLID,
          ^
/home/fac/f/bangerth/p/deal.II/1/dealii/source/lac/sparsity_tools.cc:196:10: warning: unused parameter ‘sizeLID’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/source/lac/sparsity_tools.cc:196:10: warning: unused parameter ‘globalID’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/source/lac/sparsity_tools.cc:196:10: warning: unused parameter ‘localID’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/source/lac/sparsity_tools.cc:196:10: warning: unused parameter ‘num_edges’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/source/lac/sparsity_tools.cc:196:10: warning: unused parameter ‘wgt_dim’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/source/lac/sparsity_tools.cc:196:10: warning: unused parameter ‘ewgts’ [-Wunused-parameter]
[261/772] Building CXX object source/lac/CMakeFiles/obj_lac_release.dir/lapack_full_matrix.cc.o
/home/fac/f/bangerth/p/deal.II/1/dealii/source/lac/lapack_full_matrix.cc: In instantiation of ‘void dealii::{anonymous}::cholesky_rank1(dealii::LAPACKFullMatrix<std::complex<_Tp> >&, std::complex<_Tp>, const dealii::Vector<std::complex<_Tp> >&) [with number = double]’:
/home/fac/f/bangerth/p/deal.II/1/dealii/source/lac/lapack_full_matrix.cc:367:33:   required from ‘void dealii::LAPACKFullMatrix<number>::rank1_update(number, const dealii::Vector<OtherNumber>&) [with number = std::complex<double>]’
source/lac/lapack_full_matrix.inst:20:16:   required from here
/home/fac/f/bangerth/p/deal.II/1/dealii/source/lac/lapack_full_matrix.cc:323:8: warning: unused parameter ‘A’ [-Wunused-parameter]
   void cholesky_rank1(LAPACKFullMatrix<std::complex<number>> &A,
        ^
/home/fac/f/bangerth/p/deal.II/1/dealii/source/lac/lapack_full_matrix.cc:323:8: warning: unused parameter ‘a’ [-Wunused-parameter]
/home/fac/f/bangerth/p/deal.II/1/dealii/source/lac/lapack_full_matrix.cc:323:8: warning: unused parameter ‘v’ [-Wunused-parameter]
```

--- a/include/deal.II/lac/lapack_templates.h
+++ b/include/deal.II/lac/lapack_templates.h
@@ -423,9 +423,9 @@ gemv (const char *, const types::blas_int *, const types::blas_int *, const floa
 /// Template wrapper for LAPACK functions dtrmv and strmv
 template <typename number>
 inline void
-trmv (const char *uplo, const char *trans, const char *diag,
-      const types::blas_int *N, const number *A, const types::blas_int *lda,
-      number *x, const types::blas_int *incx)
+trmv (const char * /*uplo*/, const char * /*trans*/, const char * /*diag*/,
+      const types::blas_int * /*N*/, const number * /*A*/, const types::blas_int * /*lda*/,
+      number * /*x*/, const types::blas_int * /*incx*/)
 {
   Assert (false, ExcNotImplemented());
 }
@@ -553,10 +553,10 @@ potrf (const char *uplo, const types::blas_int *n, float *A, const types::blas_i
 /// Tempalte wrapper for trcon
 template <typename number>
 inline void
-trcon(const char *norm, const char *uplo, const char *diag,
-      const types::blas_int *n, const number *A, const types::blas_int *lda,
-      number *rcond,
-      number *work, types::blas_int *iwork, types::blas_int *info)
+trcon(const char * /*norm*/, const char * /*uplo*/, const char * /*diag*/,
+      const types::blas_int * /*n*/, const number * /*A*/, const types::blas_int * /*lda*/,
+      number * /*rcond*/,
+      number * /*work*/, types::blas_int * /*iwork*/, types::blas_int * /*info*/)
 {
   Assert (false, ExcNotImplemented());
 }

--- a/source/lac/lapack_full_matrix.cc
+++ b/source/lac/lapack_full_matrix.cc
@@ -320,9 +320,9 @@ namespace
 
 
   template <typename number>
-  void cholesky_rank1(LAPACKFullMatrix<std::complex<number>> &A,
-                      const std::complex<number> a,
-                      const Vector<std::complex<number>> &v)
+  void cholesky_rank1(LAPACKFullMatrix<std::complex<number> > &/*A*/,
+                      const std::complex<number> /*a*/,
+                      const Vector<std::complex<number> > &/*v*/)
   {
     AssertThrow(false, ExcNotImplemented());
   }

--- a/source/lac/sparsity_tools.cc
+++ b/source/lac/sparsity_tools.cc
@@ -150,9 +150,9 @@ namespace SparsityTools
     }
 
 
-    void get_object_list(void *data, int sizeGID, int sizeLID,
+    void get_object_list(void *data, int /*sizeGID*/, int /*sizeLID*/,
                          ZOLTAN_ID_PTR globalID, ZOLTAN_ID_PTR localID,
-                         int wgt_dim, float *obj_wgts, int *ierr)
+                         int /*wgt_dim*/, float * /*obj_wgts*/, int *ierr)
     {
       SparsityPattern *graph = reinterpret_cast<SparsityPattern *>(data);
       *ierr = ZOLTAN_OK;
@@ -171,9 +171,9 @@ namespace SparsityTools
     }
 
 
-    void get_num_edges_list(void *data, int sizeGID, int sizeLID,
+    void get_num_edges_list(void *data, int /*sizeGID*/, int /*sizeLID*/,
                             int num_obj,
-                            ZOLTAN_ID_PTR globalID, ZOLTAN_ID_PTR localID,
+                            ZOLTAN_ID_PTR globalID, ZOLTAN_ID_PTR /*localID*/,
                             int *numEdges, int *ierr)
     {
       SparsityPattern *graph = reinterpret_cast<SparsityPattern *>(data);
@@ -193,11 +193,11 @@ namespace SparsityTools
 
 
 
-    void get_edge_list(void *data, int sizeGID, int sizeLID,
-                       int num_obj, ZOLTAN_ID_PTR globalID, ZOLTAN_ID_PTR localID,
-                       int *num_edges,
+    void get_edge_list(void *data, int /*sizeGID*/, int /*sizeLID*/,
+                       int num_obj, ZOLTAN_ID_PTR /*globalID*/, ZOLTAN_ID_PTR /*localID*/,
+                       int * /*num_edges*/,
                        ZOLTAN_ID_PTR nborGID, int *nborProc,
-                       int wgt_dim, float *ewgts, int *ierr)
+                       int /*wgt_dim*/, float * /*ewgts*/, int *ierr)
     {
       SparsityPattern *graph = reinterpret_cast<SparsityPattern *>(data);
       *ierr = ZOLTAN_OK;
@@ -232,7 +232,9 @@ namespace SparsityTools
       // Make sure that ZOLTAN is actually
       // installed and detected
 #ifndef DEAL_II_TRILINOS_WITH_ZOLTAN
-      (void)sparsity_pattern;
+      (void) sparsity_pattern;
+      (void) n_partitions;
+      (void) partition_indices;
       AssertThrow (false, ExcZOLTANNotInstalled());
 #else
 
@@ -354,7 +356,8 @@ namespace SparsityTools
     // Make sure that ZOLTAN is actually
     // installed and detected
 #ifndef DEAL_II_TRILINOS_WITH_ZOLTAN
-    (void)sparsity_pattern;
+    (void) sparsity_pattern;
+    (void) color_indices;
     AssertThrow (false, ExcZOLTANNotInstalled());
     return 0;
 #else
@@ -398,6 +401,7 @@ namespace SparsityTools
                         global_ids.data(),
                         color_exp.data());
 
+    (void) rc;
     //Check for error code
     Assert (rc == ZOLTAN_OK ,
             ExcInternalError());

